diff --git a/src/config/extend.js b/src/config/extend.js
index 80a472dab22bc4b2929066670e673f0e2cb91cc7..44877635a09cd4724b0dcfd3290971aae4b73c02 100644
--- a/src/config/extend.js
+++ b/src/config/extend.js
@@ -1,6 +1,5 @@
 const fetch = require('node-fetch');
 const Model = require('think-model');
-const Mongo = require('think-mongo');
 
 const { isNetlify, netlifyFunctionPrefix } = require('./netlify');
 
@@ -8,7 +7,6 @@ const isDeta = think.env === 'deta' || process.env.DETA_RUNTIME === 'true';
 
 module.exports = [
   Model(think.app),
-  Mongo(think.app),
   {
     context: {
       get serverURL() {
diff --git a/src/service/akismet.js b/src/service/akismet.js
index 1cc7389d0eea49914d655e6e88d0b2ce697b721e..52bfc23c426c9eba3e481e6d3c310964722f552e 100644
--- a/src/service/akismet.js
+++ b/src/service/akismet.js
@@ -13,30 +13,29 @@ module.exports = function (comment, blog) {
     return Promise.resolve(false);
   }
 
-  return new Promise(function (resolve, reject) {
-    const akismet = Akismet.client({ blog, apiKey: AKISMET_KEY });
+  const akismet = Akismet.client({ blog, apiKey: AKISMET_KEY, port: 443 });
 
-    akismet.verifyKey(function (err, verifyKey) {
-      if (err) {
-        return reject(err);
-      } else if (!verifyKey) {
-        return reject(new Error('Akismet API_KEY verify failed!'));
+  return akismet
+    .verifyKey()
+    .then(({ err: verifyErr, isValid }) => {
+      if (verifyErr) {
+        return Promise.reject(verifyErr);
+      } else if (!isValid) {
+        return Promise.reject(new Error('Akismet API_KEY verify failed!'));
       }
-
-      akismet.checkComment(
-        {
-          user_ip: comment.ip,
-          permalink: SITE_URL + comment.url,
-          comment_author: comment.nick,
-          comment_content: comment.comment,
-        },
-        function (err, spam) {
-          if (err) {
-            return reject(err);
-          }
-          resolve(spam);
-        },
-      );
+    })
+    .then(() =>
+      akismet.checkComment({
+        user_ip: comment.ip,
+        permalink: SITE_URL + comment.url,
+        comment_author: comment.nick,
+        comment_content: comment.comment,
+      })
+    )
+    .then(({ err: checkErr, isSpam }) => {
+      if (checkErr) {
+        return Promise.reject(err);
+      }
+      return isSpam;
     });
-  });
 };
diff --git a/src/service/storage/cloudbase.js b/src/service/storage/cloudbase.js
deleted file mode 100644
index a32a9ebd2e5a66ff38228458c41fd2e00134afd6..0000000000000000000000000000000000000000
diff --git a/src/service/storage/deta.js b/src/service/storage/deta.js
deleted file mode 100644
index d8a83435852661fdbc069904a002929ea5d3e3a6..0000000000000000000000000000000000000000
diff --git a/src/service/storage/leancloud.js b/src/service/storage/leancloud.js
deleted file mode 100644
index bb52b92b1ccb91d7e0f479e4ab070295f8fe9f29..0000000000000000000000000000000000000000
diff --git a/src/service/storage/mongodb.js b/src/service/storage/mongodb.js
deleted file mode 100644
index 975203dd28622a7017ec37fdae5accb49886d628..0000000000000000000000000000000000000000
diff --git a/src/service/storage/postgresql.js b/src/service/storage/postgresql.js
deleted file mode 100644
index 6b9d78aa665643e693afd06975cf581d9c7145d2..0000000000000000000000000000000000000000
diff --git a/src/service/storage/tidb.js b/src/service/storage/tidb.js
deleted file mode 100644
index 24444ae8e21cc063f22c97360901272431635fe6..0000000000000000000000000000000000000000
