diff --git a/src/config/extend.js b/src/config/extend.js
index cf939611277d45da617a8337d8181a32f9278601..40b9f6f2c66b314553a857554674698ed2ce5a62 100644
--- a/src/config/extend.js
+++ b/src/config/extend.js
@@ -1,5 +1,4 @@
 const Model = require('think-model');
-const Mongo = require('think-mongo');
 
 const { isNetlify, netlifyFunctionPrefix } = require('./netlify');
 
@@ -7,7 +6,6 @@ const isDeta = think.env === 'deta' || process.env.DETA_RUNTIME === 'true';
 
 module.exports = [
   Model(think.app),
-  Mongo(think.app),
   {
     context: {
       get serverURL() {
diff --git a/src/service/akismet.js b/src/service/akismet.js
index 1cc7389d0eea49914d655e6e88d0b2ce697b721e..52bfc23c426c9eba3e481e6d3c310964722f552e 100644
--- a/src/service/akismet.js
+++ b/src/service/akismet.js
@@ -13,30 +13,29 @@ module.exports = function (comment, blog) {
     return Promise.resolve(false);
   }
 
-  return new Promise(function (resolve, reject) {
-    const akismet = Akismet.client({ blog, apiKey: AKISMET_KEY });
+  const akismet = Akismet.client({ blog, apiKey: AKISMET_KEY, port: 443 });
 
-    akismet.verifyKey(function (err, verifyKey) {
-      if (err) {
-        return reject(err);
-      } else if (!verifyKey) {
-        return reject(new Error('Akismet API_KEY verify failed!'));
+  return akismet
+    .verifyKey()
+    .then(({ err: verifyErr, isValid }) => {
+      if (verifyErr) {
+        return Promise.reject(verifyErr);
+      } else if (!isValid) {
+        return Promise.reject(new Error('Akismet API_KEY verify failed!'));
       }
-
-      akismet.checkComment(
-        {
-          user_ip: comment.ip,
-          permalink: SITE_URL + comment.url,
-          comment_author: comment.nick,
-          comment_content: comment.comment,
-        },
-        function (err, spam) {
-          if (err) {
-            return reject(err);
-          }
-          resolve(spam);
-        },
-      );
+    })
+    .then(() =>
+      akismet.checkComment({
+        user_ip: comment.ip,
+        permalink: SITE_URL + comment.url,
+        comment_author: comment.nick,
+        comment_content: comment.comment,
+      })
+    )
+    .then(({ err: checkErr, isSpam }) => {
+      if (checkErr) {
+        return Promise.reject(err);
+      }
+      return isSpam;
     });
-  });
 };
diff --git a/src/service/markdown/index.js b/src/service/markdown/index.js
index 4cc808b7ad03089c4a213af5a918cf0811a5431b..e6a8273399721e1be61a07b763fff6463e772e5f 100644
--- a/src/service/markdown/index.js
+++ b/src/service/markdown/index.js
@@ -5,7 +5,6 @@ const MarkdownIt = require('markdown-it');
 const emojiPlugin = require('markdown-it-emoji');
 
 const { resolveHighlighter } = require('./highlight.js');
-const { mathjaxPlugin } = require('./mathjax.js');
 const { sanitize } = require('./xss.js');
 
 const getMarkdownParser = () => {
@@ -31,7 +30,7 @@ const getMarkdownParser = () => {
     html: true,
   });
 
-  const { emoji, tex, mathjax, katex, sub, sup } = plugin;
+  const { emoji, tex, katex, sub, sup } = plugin;
 
   // parse emoji
   if (emoji !== false) {
@@ -55,7 +54,7 @@ const getMarkdownParser = () => {
       output: 'mathml',
     });
   } else if (tex !== false) {
-    markdownIt.use(mathjaxPlugin, mathjax);
+    throw new Error(`tex plugin "${tex}" is not supported`);
   }
 
   return (content) => sanitize(markdownIt.render(content));
diff --git a/src/service/markdown/mathjax.js b/src/service/markdown/mathjax.js
deleted file mode 100644
index 24007229b371a991f9297527af387f2f32a507eb..0000000000000000000000000000000000000000
diff --git a/src/service/storage/cloudbase.js b/src/service/storage/cloudbase.js
deleted file mode 100644
index a32a9ebd2e5a66ff38228458c41fd2e00134afd6..0000000000000000000000000000000000000000
diff --git a/src/service/storage/deta.js b/src/service/storage/deta.js
deleted file mode 100644
index 7fba3736b9c53e7c1a01557050e2c9858ba6db53..0000000000000000000000000000000000000000
diff --git a/src/service/storage/leancloud.js b/src/service/storage/leancloud.js
deleted file mode 100644
index bb52b92b1ccb91d7e0f479e4ab070295f8fe9f29..0000000000000000000000000000000000000000
diff --git a/src/service/storage/mongodb.js b/src/service/storage/mongodb.js
deleted file mode 100644
index 975203dd28622a7017ec37fdae5accb49886d628..0000000000000000000000000000000000000000
diff --git a/src/service/storage/postgresql.js b/src/service/storage/postgresql.js
deleted file mode 100644
index df5a6ccfd0a06d077747ea67cdedf8b57b4f1670..0000000000000000000000000000000000000000
diff --git a/src/service/storage/tidb.js b/src/service/storage/tidb.js
deleted file mode 100644
index 24444ae8e21cc063f22c97360901272431635fe6..0000000000000000000000000000000000000000
